File: src/main/java/com/nurkiewicz/tsclass/parser/ast/expr/NumberLiteral.java
package com.nurkiewicz.tsclass.parser.ast.expr;

import lombok.Value;

@Value
public class NumberLiteral implements Expression {
    private final double value;

    public static NumberLiteral num(double x) {
        return new NumberLiteral(x);
    }

    public String toString() {
        return String.valueOf(value);
    }
}

File: src/main/kotlin/com/nurkiewicz/tsclass/parser/ast/expr/NumberLiteral.kt
package com.nurkiewicz.tsclass.parser.ast.expr

data class NumberLiteral(val value: Double) : Expression {

    override fun toString() = value.toString()

    companion object {

        @JvmStatic
        fun num(x: Double): NumberLiteral {
            return NumberLiteral(x)
        }
    }
}
